docs/api/session.md

```diff
@@ -143,6 +143,71 @@ Returns:
 Emitted after an extension is loaded and all necessary browser state is
 initialized to support the start of the extension's background page.
 
+#### Event: 'file-system-access-restricted'
+
+Returns:
+
+* `event` Event
+* `details` Object
+  * `origin` string - The origin that initiated access to the blocked path.
+  * `isDirectory` boolean - Whether or not the path is a directory.
+  * `path` string - The blocked path attempting to be accessed.
+* `callback` Function
+  * `action` string - The action to take as a result of the restricted path access attempt.
+    * `block` - This will block the access request and trigger an [`AbortError`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort).
```

Comment:
[setWindowOpenHandler](https://www.electronjs.org/docs/latest/api/web-contents#contentssetwindowopenhandlerhandler) uses allow/deny. What do you think about changing to this?

Reply:
@samuelmaddock I'm not sure I see why we'd want to remove granularity in possible behaviors given all three potential choices have clear use cases for developers - allow/deny makes sense in a window open context but here I don't think a binary choice is what we want.

Reply:
Oh sorry, I meant allow/deny/tryAgain

Reply:
ohh 😅 misparsed - yes, i think you're right, i'll change that!